### PR TITLE
Add a note that changing `codec` only applies to newly written segments

### LINKED ISSUE #12629: Changing `codec` on existing partitions appears to have no effect
### CrateDB version

4.8.0

### CrateDB setup information

_No response_

### Steps to Reproduce

1. Create a partitioned table:
    ```sql
    CREATE TABLE IF NOT EXISTS "tmp"."conditions" (
          "time" TIMESTAMP WITH TIME ZONE NOT NULL,
          "device_id" TEXT,
          "temperature" DOUBLE PRECISION,
          "humidity" DOUBLE PRECISION,
          "partition_key" INTEGER
     )
     PARTITIONED BY ("partition_key");
    ```
2. Insert some data to create a partition (should be enough data to notice the effect of compression later on)
3. Change the partitions' codec:
    ```sql
    ALTER TABLE tmp.conditions PARTITION (partition_key = 1) CLOSE;
    ALTER TABLE tmp.conditions PARTITION (partition_key = 1) SET (codec = 'best_compression');
    ALTER TABLE tmp.conditions PARTITION (partition_key = 1) OPEN;
    ```
4. Verify the codec was changed:
     ```sql
     SELECT settings['codec']
     FROM information_schema.table_partitions
     WHERE table_schema = 'tmp'
       AND table_name = 'conditions';
    ```
    Returns: `best_compression` 
5. To be on the safe side (not sure if needed), also run `OPTIMIZE TABLE tmp.conditions`

*Observation*
After changing `codec`, the table size doesn't decrease. Creating a copy of the same data with `codec = 'best_compression'` set during `CREATE TABLE` shows that the partition size is noticeably smaller if compression is applied successfully. Therefore, it seems as if compression with the workflow above isn't actually changed.

Also see the related [community discussion](https://community.crate.io/t/optimizing-storage-for-historic-time-series-data/762/4).

### Expected Result

Not sure if changing the codec on an existing partition is supposed to work, but I would expect one of the following outcomes:
1. The partition size does decrease
2. There is an error message stating that changing `codec` for existing partitions isn't supported

### Actual Result

Compression doesn't change, even after waiting for a few hours.

### PR DESCRIPTION
**Pull request by hammerhead**
## Summary of the changes / Why this improves CrateDB
Fixes #12629 by mentioning the need to run `OPTIMIZE` when changing `codec`, potentially with `max_num_segments = 1` if the table was already optimized.

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [X] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Review by seut — APPROVED** [review]
Thanks!

**Review by matriv — APPROVED** [review]
Thank you!

File: docs/general/ddl/alter-table.rst [hunk 1]
```diff
@@ -32,6 +32,10 @@ In order to set a parameter to its default value use ``reset``::
     cr> alter table my_table reset (number_of_replicas);
     ALTER OK, -1 rows affected (... sec)
 
+.. NOTE::
+    Changing compression using the ``codec`` parameter only takes effect for
+    newly written segments. To enforce a rewrite of already existing segments,
+    run :ref:`sql-optimize` with the parameter ``max_num_segments = 1``.
```
Comment (amotl) [inline docs/general/ddl/alter-table.rst:38]: I think a newline is needed after the admonition identifier. Not 100% sure though.

Comment (amotl) [reply docs/general/ddl/alter-table.rst:38]: GitHub renders it differently than Sphinx, but there _is_ a difference: https://gist.github.com/amotl/3c6085f1b0972a826a3fa7dddd94d5e2

Comment (hammerhead) [reply docs/general/ddl/alter-table.rst:38]: I tested it locally using `./gradlew :blackbox:developDocs`, appears to render fine:
Screenshot 2022-06-22 at 08 21 31

**Pull request merged**